### ISSUE TITLE
Create-AadAppsForBC no longer accepting local hostnames.

### ISSUE THREAD

**Issue by vvkjndl** [microsoft/navcontainerhelper#2251]
**Describe the issue**
Hi Freddy, it looks like a change has been introduced recently in the underlying command `New-AzureADApplication` that we use within the function `Create-AadAppsForBC`.

Here's an excerpt on faulting parameter from `New-AzureADApplication` help page:
```
-IdentifierUris <System.Collections.Generic.List`1[System.String]>
        User-defined URI(s) that uniquely identify a Web application within its Azure AD tenant, or within a verified custom domain (see "Domains" tab in the Azure classic portal) if the application is multi-tenant.

        The first element is populated from the Web application's "APP ID URI" field if updated via the Azure classic portal (or respective Azure AD PowerShell cmdlet parameter). Additional URIs can be added via the application manifest; see Understanding the Azure AD
        Application Manifest for details. This collection is also used to populate the Web application's servicePrincipalNames collection.
```

**Scripts used to create container and cause the issue**
```
Create-AadAppsForBC -AadAdminCredential $azadmincred -appIdUri "https://vjaad/BC/" -publicWebBaseUrl "https://vjaad/BC/" -IncludeApiAccess
```

**Full output of scripts**
```
Creating AAD App for WebClient
Create-AadAppsForBC Telemetry Correlation Id: 4ab264b2-5e42-48eb-9958-ca960f1639f5
New-AzureADApplication : Error occurred while executing NewApplication
Code: Request_BadRequest
Message: Values of identifierUris property must use a verified domain of the organization or its subdomain: 'https://vjaad.local/BC/'
RequestId: b5a76c85-53e8-40db-ade4-c458ed7eb91d
DateTimeStamp: Tue, 04 Jan 2022 13:46:44 GMT
Details: PropertyName  - identifierUris, PropertyErrorCode  - HostNameNotOnVerifiedDomain, HostName  - https://vjaad.local/BC/
HttpStatusCode: BadRequest
HttpStatusDescription: Bad Request
HttpResponseStatus: Completed
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.0\AzureAD\Create-AadAppsForNav.ps1:122 char:17
+ ... $ssoAdApp = New-AzureADApplication -DisplayName "WebClient for $appId ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [New-AzureADApplication], ApiException
    + FullyQualifiedErrorId : Microsoft.Open.AzureAD16.Client.ApiException,Microsoft.Open.AzureAD16.PowerShell.NewApplicatio
```

**Screenshots**
N/A.

**Additional context**
- does it happen all the time?
Yes.
- did it use to work?
Yes.

**Comment by freddydk** [comment]
That is correct and there is very little I can do about that.
The breaking change introduced by Azure in October is described here: https://docs.microsoft.com/en-us/azure/active-directory/develop/reference-breaking-changes#appid-uri-in-single-tenant-applications-will-require-use-of-default-scheme-or-verified-domains

The fix applied to the Azure VMs (https://aka.ms/getbc) is here: https://github.com/microsoft/nav-arm-templates/commit/c8813c86d0abbeae48c1b659a39d1635d418af34

Basically using the domain of the Office365UserName together with the containername (or host name)

**Comment by vvkjndl** [comment]
Unfortunately this is not our use case here. We are running a local container and exposing it through a tunnel.

So the real URL returned by BcContainer will still be: `https://vjaad/BC/SignIn`

**Comment by freddydk** [comment]
Correct - but you should set the appiduri to be https://vjadd.<the domain from your office 365 creds) - both when creating the aad app and the container.

It should work - will try myself in the near future and blog about it.

**Comment by vvkjndl** [comment]
Makes sense now.

**Comment by vvkjndl** [comment]
@freddydk Creating container is not allowing dots.

`Container name contains invalid characters. Allowed characters are letters (a-z), numbers (0-9) and dashes (-)`

**Comment by freddydk** [comment]
ContainerName should still be the original one.
I will look into some samples when I am over my Covid-19 infection.

**Comment by vvkjndl** [comment]
@freddydk Sure freddydk and best wishes from ours side on getting recovered soon.

**Comment by vvkjndl** [comment]
Thanks a lot freddy for doing this especially given the circumstances.
